src/fundus/scraping/pipeline.py

```diff
@@ -1,47 +1,163 @@
-                    yield next(robin)
-                except StopIteration:
-                    pass
-                max_articles -= 1
+        """Yields articles from initialized scrapers.
+
+        The articles will be requested concurrently with batch size := len(self.scrapers).
+        You can specify a Delay to be used between the batches with <delay>
+
+        Works like a light-wight version of Crawler.crawl(). Ment to be used when dealing with
+        custom scrapers outside the context of predefined publisher collections.
+        Refer to the docstring of Crawler.crawl() for more detailed information about the Args.
+
+        Args:
+            max_articles (Optional[int]): Maximal number of articles to be yielded. Defaults to None.
+            error_handling (Literal["suppress", "catch", "raise"]): Set error handling
+                for extraction. Defaults to "suppress".
+            extraction_filter (Optional[ExtractionFilter]): Set extraction filter. Defaults to None.
+            delay (Optional[Delay]): Set waiting time between article batches. Defaults to None.
+            url_filter (Optional[URLFilter]): Set URLFilter. Defaults to None
+            only_unique: (bool): If true return only unique responses. Defaults to True.
+
+        Returns:
+            Iterator[Article]: An iterator yielding objects of type Article.
+        """
+
+        response_cache: Set[str] = set()
+
+        for scraper in self.scrapers:
+            for source in scraper.sources:
+                if url_filter:
+                    source.add_url_filter(url_filter=url_filter)
+                if only_unique:
+                    source.add_url_filter(url_filter=lambda url: url in response_cache)
+
+        scrape_map = [
+            scraper.scrape(
+                error_handling=error_handling,
+                extraction_filter=extraction_filter,
+            )
+            for scraper in self.scrapers
+        ]
+
+        loop = asyncio.get_event_loop()
+
+        def article_gen() -> Iterator[Article]:
+            interleave: AsyncIterator[Iterable[Optional[Article]]] = batched_async_interleave(*list(scrape_map))
+            while True:
+                start_time = time.time()
+                batch: Optional[Iterable[Optional[Article]]] = loop.run_until_complete(async_next(interleave, None))
+                batch_time = time.time() - start_time
+                if delay:
+                    actual_delay = max(delay() - batch_time, 0.0)
+                    loop.run_until_complete(asyncio.sleep(actual_delay))
+                basic_logger.debug(f"Batch took {batch_time} seconds")
+                if batch is not None:
+                    for next_article in cast(Iterable[Article], filter(bool, batch)):
+                        if only_unique:
+                            response_cache.add(next_article.html.responded_url)
+                        yield next_article
+                else:
+                    break
```

Comment:
I think that is really great idea :) Thanks for sharing your thoughts.

I'm undecided about where to continue from now:

- On the one hand, the feature you're describing should definitely come to Fundus, at best with this PR.
- But on the other hand, this will take some major rewriting on the pipeline logic and definitely take some time.

@dobbersc what do you think? Postpone it to another PR so this one gets merged quicker but with the knowledge that it has to be rewritten, or postpone this PR, but then the release and docs have to wait. 

I think I would favor release over completeness right now. Would you be fine with postponing your requests?

Reply:
Let's discuss this in tomorrow's meeting.

Reply:
We decided to do this in another PR.